### PR TITLE
frontend tests for settings screens

### PR DESCRIPTION
**Pull request by Nocccer**
frontend tests for src/screends/Settings:

- [x] GeneralSettings
- [x] OtherSettings
- [x] Tools
- [x] SyncSaves
- [x] WineSettings
- [x] the whole settings screen
- [x] FixMe: replaceAll not working with jest.
- [x] mock i18next globally
- [x] add template class to setup test variables easier

### CODE REVIEW

**Comment by Nocccer** [comment]
Please have a look more on the source files instead of test files. I changed some minor stuff there.

File: .vscode/launch.json [hunk 1]
```diff
@@ -1,6 +1,12 @@
 {
   "version": "0.2.0",
   "configurations": [
+    {
```
Comment (flavioislima) [inline .vscode/launch.json:4]: What that does?

File: src/screens/Settings/components/SyncSaves/index.tsx [hunk 2]
```diff
@@ -90,36 +90,37 @@ export default function SyncSaves({
...
+            data-testid="inputSavePath"
             type="text"
             placeholder={t('setting.savefolder.placeholder')}
             className="settingSelect"
             value={savesPath}
             disabled={isSyncing}
             onChange={(event) => setSavesPath(event.target.value)}
           />
-          {!savesPath.length ? (
+          {!isLinked ? (
```
Comment (flavioislima) [inline src/screens/Settings/components/SyncSaves/index.tsx:105]: Good catch since isLinked is doing the same thing as savesPath.length

File: src/screens/Settings/components/Tools/index.tsx [hunk 3]
```diff
@@ -47,9 +47,10 @@ export default function Tools({ wineVersion, winePrefix }: Props) {
       // Use DataTransferItemList interface to access the file(s)
       // If dropped items aren't files, reject them
       if (ev.dataTransfer.items[0].kind === 'file') {
-        const exe = ev?.dataTransfer?.items[0]?.getAsFile()?.path
-
-        return callTools('runExe', exe)
+        const exe = ev.dataTransfer.items[0].getAsFile()?.path
```
Comment (flavioislima) [inline src/screens/Settings/components/Tools/index.tsx:50]: What happens if `dataTransfer` comes empty?

File: src/screens/Settings/components/WineSettings/index.tsx [hunk 4]
```diff
@@ -74,21 +74,23 @@ export default function WineSettings({
   function removeCustomPath() {
     const newPaths = customWinePaths.filter((path) => path !== selectedPath)
     setCustomWinePaths(newPaths)
-    return setSelectedPath(customWinePaths.length ? customWinePaths[0] : '')
+    return setSelectedPath(newPaths.length ? newPaths[0] : '')
```
Comment (flavioislima) [inline src/screens/Settings/components/WineSettings/index.tsx:77]: why the change?

**Review by flavioislima — COMMENTED** [review]
Looks really good, this is a phenomenal job you did here, man!
Left just a few comments for some things I didn't understand but other than that its really amazing :D

File: .vscode/launch.json (hunk 1 above)
Comment (Nocccer) [reply .vscode/launch.json:4]: Just a way for me to debug the tests by pressing f5. Can remove that, if not needed.

File: src/screens/Settings/components/WineSettings/index.tsx (hunk 4 above)
Comment (Nocccer) [reply src/screens/Settings/components/WineSettings/index.tsx:77]: Year now i see this can lead to problems. Will revert that and adapt the test.

File: src/screens/Settings/components/Tools/index.tsx (hunk 3 above)
Comment (Nocccer) [reply src/screens/Settings/components/Tools/index.tsx:50]: check the if before:  
`if( env.dataTransfer.items)`
We already catched that.

File: .vscode/launch.json (hunk 1 above)
Comment (flavioislima) [reply .vscode/launch.json:4]: Looks nice, keep it!

File: src/screens/Settings/components/Tools/index.tsx (hunk 3 above)
Comment (flavioislima) [reply src/screens/Settings/components/Tools/index.tsx:50]: ok then

File: src/screens/Settings/components/WineSettings/index.tsx (hunk 4 above)
Comment (flavioislima) [reply src/screens/Settings/components/WineSettings/index.tsx:77]: ok

File: src/screens/Settings/components/Tools/index.tsx (hunk 3 above)
Comment (Nocccer) [reply src/screens/Settings/components/Tools/index.tsx:50]: i found that while testing. I could never reach the point, because i can't manipulate the dataTransfer after the if statement in time.

File: src/screens/Settings/components/WineSettings/index.tsx (hunk 4 above)
Comment (Nocccer) [reply src/screens/Settings/components/WineSettings/index.tsx]: Reverted.

**Review by flavioislima — APPROVED** [review]
Looks amazing. Good job!
